src/Fixer/ReturnNotation/SimplifiedNullReturnFixer.php

```diff
@@ -101,21 +102,48 @@ private function clear(Tokens $tokens, $index)
      */
     private function needFixing(Tokens $tokens, $index)
     {
-        $content = '';
+        if ($this->isNullableReturnTypeFunction($tokens, $index)) {
+            return false;
+        }
 
+        $content = '';
         while (!$tokens[$index]->equals(';')) {
             $index = $tokens->getNextMeaningfulToken($index);
             $content .= $tokens[$index]->getContent();
         }
 
-        $content = rtrim($content, ';');
         $content = ltrim($content, '(');
-        $content = rtrim($content, ')');
+        $content = rtrim($content, ');');
 
         return 'null' === strtolower($content);
     }
 
     /**
+     * Is the return within a function with a nullable return type?
+     *
+     * @param Tokens $tokens
+     * @param int    $index
```

Comment:
this is an index of ?

Reply:
Just copied from the other fixer methods. Btw I've added the  description now.